include/swift/Basic/BlotMapVector.h

```diff
@@ -85,27 +88,34 @@ class BlotMapVector {
     }
 
     const_iterator find(const KeyT &Key) const {
-      return const_cast<BlotMapVector &>(*this)->find(Key);
+      return const_cast<BlotMapVector &>(*this).find(Key);
     }
 
-    /// This is similar to erase, but instead of removing the element from the
-    /// vector, it just zeros out the key in the vector. This leaves iterators
-    /// intact, but clients must be prepared for zeroed-out keys when iterating.
-    void erase(const KeyT &Key) { blot(Key); }
-
-    /// This is similar to erase, but instead of removing the element from the
-    /// vector, it just zeros out the key in the vector. This leaves iterators
+    /// Eliminate the element at `Key`. Instead of removing the element from the
+    /// vector, just zero out the key in the vector. This leaves iterators
     /// intact, but clients must be prepared for zeroed-out keys when iterating.
+    ///
+    /// Return true if the element was erased.
+    bool erase(const KeyT &Key) { return blot(Key); }
+
+    /// Eliminate the element at the given iterator. Instead of removing the
+    /// element from the vector, just zero out the key in the vector. This
+    /// leaves iterators intact, but clients must be prepared for zeroed-out
+    /// keys when iterating.
     void erase(iterator I) { erase((*I)->first); }
 
-    /// This is similar to erase, but instead of removing the element from the
+    /// Eliminate the element at `Key`. Instead of removing the element from the
     /// vector, it just zeros out the key in the vector. This leaves iterators
     /// intact, but clients must be prepared for zeroed-out keys when iterating.
-    void blot(const KeyT &Key) {
+    ///
+    /// Return true if the element was found and erased.
+    bool blot(const KeyT &Key) {
```

Comment:
Maybe just move this into erase? I don't remember why it is separate.

Reply:
I don't understand why it's separate but wasn't terribly interested in changing all the clients in this PR.

Reply:
Ok. If it isn't too much work, can you do it in a follow on commit? Or I may do it in a drive by change later.